x/params/keeper.go

```diff
@@ -0,0 +1,328 @@
+package params
+
+import (
+	"fmt"
+	"reflect"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/wire"
+)
+
+// Keeper manages global parameter store
+type Keeper struct {
+	cdc *wire.Codec
+	key sdk.StoreKey
+}
+
+// NewKeeper constructs a new Keeper
+func NewKeeper(cdc *wire.Codec, key sdk.StoreKey) Keeper {
+	return Keeper{
+		cdc: cdc,
+		key: key,
+	}
+}
+
+// InitKeeper constructs a new Keeper with initial parameters
+func InitKeeper(ctx sdk.Context, cdc *wire.Codec, key sdk.StoreKey, params ...interface{}) Keeper {
+	if len(params)%2 != 0 {
+		panic("Odd params list length for InitKeeper")
+	}
+
+	k := NewKeeper(cdc, key)
+
+	for i := 0; i < len(params); i += 2 {
+		k.set(ctx, params[i].(string), params[i+1])
+	}
+
+	return k
+}
+
+// get automatically unmarshalls parameter to pointer
+func (k Keeper) get(ctx sdk.Context, key string, ptr interface{}) error {
+	store := ctx.KVStore(k.key)
+	bz := store.Get([]byte(key))
+	return k.cdc.UnmarshalBinary(bz, ptr)
+}
+
+// getRaw returns raw byte slice
+func (k Keeper) getRaw(ctx sdk.Context, key string) []byte {
+	store := ctx.KVStore(k.key)
+	return store.Get([]byte(key))
+}
+
+// set automatically marshalls and type check parameter
+func (k Keeper) set(ctx sdk.Context, key string, param interface{}) error {
```

Comment:
Can we have an assortment of typed `set` functions like the `GetXYZ` functions? I think we only want to allow specific types.